setup.py

```diff
@@ -1,124 +1,88 @@
-    data_roots = ['pydicom/data']
-    for data_root in data_roots:
-        for root, subfolder, files in os.walk(data_root):
-            files = [
-                x.replace('pydicom/', '') for x in glob(root + '/*')
-                if not os.path.isdir(x)
-            ]
-            files = [f for f in files if not f.endswith('.pyc')]
-            data_files += files
-    return data_files
-
-
-PACKAGE_DATA = {
-    'pydicom': data_files_inventory() + ['py.typed']
-}
-
-ENTRY_POINTS = {
-    "console_scripts": ["pydicom=pydicom.cli.main:main"],
-    "pydicom_subcommands": [
-        "codify = pydicom.cli.codify:add_subparser",
-        "show = pydicom.cli.show:add_subparser"
+    root = BASE_DIR / "pydicom" / "data"
+    files = [
+        f.relative_to(BASE_DIR / "pydicom")
+        for f in root.glob("**/*")
+        if f.is_file() and f.suffix != ".pyc"
     ]
-}
+    return [os.fspath(f) for f in files]
 
 
-opts = dict(
-    name=NAME,
-    python_requires='>=3.6.1',
-    version=VERSION,
-    maintainer=MAINTAINER,
-    maintainer_email=MAINTAINER_EMAIL,
-    author=AUTHOR,
-    author_email=AUTHOR_EMAIL,
-    description=description,
-    long_description=LONG_DESCRIPTION,
+setup(
+    name="pydicom",
+    version=__version__,
+    author="Darcy Mason and contributors",
+    author_email="darcymason@gmail.com",
+    description="A pure Python package for reading and writing DICOM data",
+    long_description=long_description,
     long_description_content_type='text/markdown',
-    url=URL,
-    download_url=DOWNLOAD_URL,
-    license=LICENSE,
-    keywords=KEYWORDS,
-    classifiers=CLASSIFIERS,
+    url="https://github.com/pydicom/pydicom",
+    download_url="https://github.com/pydicom/pydicom/archive/master.zip",
+    license="MIT",
+    keywords="dicom python medical imaging",
+    classifiers=[
+        "License :: OSI Approved :: MIT License",
+        "Intended Audience :: Developers",
+        "Intended Audience :: Healthcare Industry",
+        "Intended Audience :: Science/Research",
+        "Development Status :: 5 - Production/Stable",
+        "Programming Language :: Python",
+        "Programming Language :: Python :: 3.6",
+        "Programming Language :: Python :: 3.7",
+        "Programming Language :: Python :: 3.8",
+        "Programming Language :: Python :: 3.9",
+        "Programming Language :: Python :: 3.10",
+        "Operating System :: OS Independent",
+        "Topic :: Scientific/Engineering :: Medical Science Apps.",
+        "Topic :: Scientific/Engineering :: Physics",
+        "Topic :: Software Development :: Libraries"
+    ],
     packages=find_packages(),
-    py_modules=_py_modules,
-    package_data=PACKAGE_DATA,
+    py_modules=[] if not HAVE_DICOM else ["dicom"],
+    package_data={
+        'pydicom': data_files_inventory() + ['py.typed'],
+    },
     include_package_data=True,
-    install_requires=REQUIRES,
-    setup_requires=SETUP_REQUIRES,
-    tests_require=TESTS_REQUIRE,
```

Comment:
`setup_requires` is deprecated in favour of `pyproject.toml` and `tests_require` is just deprecated

Reply:
Are we sure that this isn't still required in some distribtution packaging environment? I know almost nothing about that, so this may be a dumb question...

Reply:
> Are we sure that this isn't still required in some distribtution packaging environment?

I echo the concerns here.  Pydicom has been through so many iterations of the latest "new right way to do it" packaging only to see that change again shortly after.  I'd stay on the trailing edge of these deprecations, only changing when the new way is long established.  Having said that, I'm just catching up via the links given, so not sure how established these changes actually are.

Reply:
> Are we sure that this isn't still required in some distribtution packaging environment?

Either way it shouldn't matter. We don't actually have any build dependencies (for `setup_requires`), the one set currently looks like it was a shortcut to let someone run `pip install pydicom tests` and get pytest installed (very much non-standard). 

`tests_require` has been deprecated since 2019 or so, plus it's incomplete; it doesn't even have numpy. 

But I can add them back in if that's what you want.

Reply:
`setup_requires` should be the packages required to build the package (although as I said, this should be done with `pyproject.toml` these days). If you have a C add-on you'd use `setup_requires` to tell `setuptools` that `cython` needs to be installed to build the add-in. So it's really not intended for installing `pytest` like this:

```python
needs_pytest = {'pytest', 'test', 'ptr'}.intersection(sys.argv)
pytest_runner = ['pytest-runner'] if needs_pytest else []
SETUP_REQUIRES = pytest_runner
```

Reply:
Also, [lol](https://pypi.org/project/pytest-runner/):
> pytest-runner depends on deprecated features of setuptools and relies on features that break security mechanisms in pip. For example ‘setup_requires’ and ‘tests_require’ bypass pip --require-hashes. See also [pypa/setuptools#1684](https://github.com/pypa/setuptools/issues/1684).
>
> It is recommended that you:
> * Remove 'pytest-runner' from your setup_requires, preferably removing the setup_requires option.
> * Remove 'pytest' and any other testing requirements from tests_require, preferably removing the tests_requires option.
```

Reply:
Ok, thanks - that clarifies it for me!

Reply:
Are you happy with this @darcymason?

Reply:
Yes, thanks for the links, etc.  Looks clear that we should move forward with these changes.